src/toil/cwl/cwltoil.py

```diff
@@ -1805,6 +1806,77 @@ def determine_load_listing(tool: ToilCommandLineTool):
         )
     return load_listing
 
+class NoAvailableJobStoreException(Exception):
+    """Indicates that no job store name is available."""
+    pass
+
+def generate_default_job_store(batch_system_name: Optional[str], provisioner_name: Optional[str],
```

Comment:
Should either of these args default to `None`?

Reply:
They can't have defaults with this argument order, I don't think; `local_directory` is a required argument.

The function always needs access to the relevant option values, even if those happen to be None.